File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/LinearSlideTest.java
package org.firstinspires.ftc.teamcode.opmodes;

import org.firstinspires.ftc.teamcode.features.TiltableLinearSlide;
import org.firstinspires.ftc.teamcode.internals.hardware.Devices;
import org.firstinspires.ftc.teamcode.internals.registration.OperationMode;
import org.firstinspires.ftc.teamcode.internals.registration.TeleOperation;

/**
 * This tests the linear slide. Uses a motor on motor 0, and the controls are the left and right triggers.
 */
public class LinearSlideTest extends OperationMode implements TeleOperation {
    @Override
    public void construct() {

    }

    @Override
    public void run() {
        Devices.motor0.setPower(Devices.controller1.getRightTrigger() - Devices.controller1.getLeftTrigger());
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/LinearSlideTest.kt
package org.firstinspires.ftc.teamcode.opmodes

import com.qualcomm.robotcore.eventloop.opmode.TeleOp
import org.firstinspires.ftc.teamcode.internals.templates.BaseOpMode
import org.firstinspires.ftc.teamcode.scripts.ContinuousMotorScript


@TeleOp(name = "Linear Slide Test", group = BaseOpMode.DEBUG_GROUP_NAME)
class LinearSlideTest : BaseOpMode() {
    override fun construct() {
        addScript(ContinuousMotorScript(id = 6, input = { (gamepad1.left_trigger - gamepad1.right_trigger).toDouble() }))
    }

    override fun run() {

    }

    override fun onStop() {

    }
}